docs/user_guide/ml.md

```diff
@@ -1,8 +1,269 @@
 # Machine Learning
 
-In this section, the users will learn how to implement machine learning in their analysis with
-columnflow.
+In this section, the users will learn how to implement machine learning in their analysis with columnflow.
 
-Note: which tasks, how to trigger them, how to train, test and plot networks and outputs
+# How training happens in Columnflow: K-fold cross validation
+Machine learning in columnflow is implemented in a way that k-fold cross validation is enabled by default.
+In k-fold cross validation the dataset is split in k-parts of equal size.
+For each training, k-1 parts are used for the actual training and the remaining part is used for validation of the model.
+This process is repeated k-times, resulting in the training of k-model instances.
+In the end of the training columnflow will save all k-models, which are then usable for evaluation.
+An overview and further details about possible variations of k-fold cross validation can be found in the [sci-kit documentation](https://scikit-learn.org/stable/modules/cross_validation.html).
 
-TODO
+# Configure your custom machine learning class:
+To create a custom machine learning (ML) class in columnflow, it is imperative to inherit from the {py:class}`~columnflow.ml.MLModel` class.
+This inheritance ensures the availability of functions to manage and access config and model instances, as well as the necessary producers.
+The name of your custom ML class can be arbitrary, since `law` accesses your machine learning model using a `cls_name` in {py::meth}`~columnflow.util.DerivableMeta.derive`, e.g.
```

Comment:
```suggestion
The name of your custom ML class can be arbitrary, since `law` accesses your machine learning model using a `cls_name` in {py:meth}`~columnflow.util.DerivableMeta.derive`, e.g.
```